File: typescript/src/Equipment.ts
import {Item} from './Item';

export class Equipment {

    // TODO add a ring item that may be equipped
    // that may also add damage modifier
    public constructor(private _leftHand: Item,
                private _rightHand: Item,
                private _head: Item,
                private _feet: Item,
                private _chest: Item) {}

    public get leftHand(): Item {
        return this._leftHand;
    }

    public get rightHand(): Item {
        return this._rightHand;
    }

    public get head(): Item {
        return this._head;
    }

    public get feet(): Item {
        return this._feet;
    }

    public get chest(): Item {
        return this._chest;
    }
}

File: go/equipment.go
package codingdojo

type (
	Equipment struct{ Items } // TODO add a ring item that may be equipped that may also add damage modifier

	Items struct {
		leftHand  Item
		rightHand Item
		head      Item
		feet      Item
		chest     Item
	}
)

func MakeEquipment(items Items) Equipment {
	return Equipment{items}
}

func (e Equipment) GetLeftHand() Item {
	return e.leftHand
}
func (e Equipment) GetRightHand() Item {
	return e.rightHand
}
func (e Equipment) GetHead() Item {
	return e.head
}
func (e Equipment) GetFeet() Item {
	return e.feet
}
func (e Equipment) GetChest() Item {
	return e.chest
}
